examples/output.rs

```diff
@@ -0,0 +1,153 @@
+        }
+        if input.just_released(*keycode) {
+            output.send([0b10000000, *note, 127]); // Note on, channel 1, max velocity
+        }
+    }
+}
+
+fn print_errors(mut errors: EventReader<MidiOutputError>) {
+    for err in errors.iter() {
+        error!("{}", err);
+    }
+}
+
+#[derive(Component)]
+pub struct Instructions;
+
+fn show_ports(output: Res<MidiOutput>, mut instructions: Query<&mut Text, With<Instructions>>) {
+    if output.is_changed() {
+        let text_section = &mut instructions.single_mut().sections[1];
+        text_section.value = "Available output ports:\n\n".to_string();
+        for (i, (name, _)) in output.ports().iter().enumerate() {
+            text_section
+                .value
+                .push_str(format!("Port {:?}: {:?}\n", i, name).as_str());
+        }
+    }
+}
+
+fn show_connection(
+    connection: Res<MidiOutputConnection>,
+    mut instructions: Query<&mut Text, With<Instructions>>,
+) {
+    if connection.is_changed() {
+        let text_section = &mut instructions.single_mut().sections[2];
+        if connection.is_connected() {
+            text_section.value = "Connected".to_string();
+            text_section.style.color = Color::GREEN;
+        } else {
+            text_section.value = "Disconnected".to_string();
+            text_section.style.color = Color::RED;
+        }
+    }
+}
+
+fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
+    commands.spawn_bundle(Camera2dBundle::default());
+
+    commands
+        .spawn_bundle(TextBundle {
+            text: Text {
+                sections: vec![
+                    TextSection::new(
+                        "INSTRUCTIONS \n\
+                    R - Refresh ports \n\
+                    A to G - Play note \n\
+                    0 to 9 - Connect to port \n\
+                    Escape - Disconnect from current port \n",
+                        TextStyle {
+                            font: asset_server.load("fonts/FiraSans-Bold.ttf"),
+                            font_size: 30.0,
+                            color: Color::WHITE,
+                        },
+                    ),
+                    TextSection::from_style(TextStyle {
+                        font: asset_server.load("fonts/FiraSans-Bold.ttf"),
+                        font_size: 30.0,
+                        color: Color::BLACK,
+                    }),
+                    TextSection::new(
+                        "Disconnected",
+                        TextStyle {
+                            font: asset_server.load("fonts/FiraSans-Bold.ttf"),
+                            font_size: 30.0,
+                            color: Color::RED,
+                        },
+                    ),
+                ],
+                alignment: TextAlignment::TOP_LEFT,
+            },
+            ..default()
+        })
+        .insert(Instructions);
+}
```

Comment:
This is nice, we should have a connection part for input as well, instead of only allowing for via the terminal

Reply:
Agreed